src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorConfirmService.java

```diff
@@ -101,6 +101,10 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDet
 
         log.info("Due date after submit Task is {}", updateDetails.getData().getDueDate());
 
+        if (caseData.getApplicationType().isSole() && caseData.getAcknowledgementOfService().getDateAosSubmitted() != null) {
```

Comment:
Same comment from the CaseworkerConfirmService event, can we use the service layer for the state logic.

Reply:
as above.

Reply:
Resolved after refactor.